### PR TITLE
initial design considerations

### PR DESCRIPTION
**Pull request by GerhardRaven**

### CODE REVIEW

File: FrameworkDesignConsiderations.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+To enable concurrency, the first hurdle seems to be the description of data dependencies
+amongst Algorithms. One option could be to add sufficient introspection to Algorithms,
+and thus allow the framework to deduce the input(s) and output(s) of each Algorithm. This
+could eg. be done by requiring all interaction with the TES to be done through an API which
+enforces the use of dedicated Properties which specify TES locations and the type of 
+access (read, write, or update). (note: update should be forbidden).
+
+An alternative approach would be to recognize that most (all?) Algorithms actually 
+follow the same pattern: read data and then transform it new data, and then make this
```
Comment (betatim) [inline FrameworkDesignConsiderations.md]: transform it new data -> transform it to new data

File: FrameworkDesignConsiderations.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+data available in the TES. As such, one could consider writing one single generic algorithm
+which wraps this functionality (and thus provides a uniform pattern), and delegate the
+computational work. This suggests a new type of Gaudi component, which is explictly provided
+with input data, and explicitly provides output data -- lets define this as a Transform..
```
Comment (betatim) [inline FrameworkDesignConsiderations.md]: Transform.. -> Transform.

**Comment by betatim** [comment]
I would turn things like "...Algorithm..." into "...`Algorithm`..." and similar for programming things (don't know a better word).

Otherwise looks good to me.

File: FrameworkDesignConsiderations.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# Design Considerations
+
+The framework will have to work efficiently on many/multi core systems.
+Presumably, this will imply some level of concurrency. As a result, scheduling
+operations to be performed on the data will become more dynamic, and much
+more complex then our current serial case, where 'Algorithm's are statically
```
Comment (bencouturier) [inline FrameworkDesignConsiderations.md]: more complex than

File: FrameworkDesignConsiderations.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+A prime example of reduction is `std::accumulate`. It operates an a vector of values,
+and can be seen as a generalization of eg. summation. This operation is eg. needed
+to combine a list of Hlt decisions into one global decision -- in this case the
+operation is logical or. Let us call the corresponding algorithm `Accumulator`. 
+
+
+
+# Concurrency Friendly
+In the above, we outlined that code should be written in a 'concurrency friendly' style.
+Here we detail want this entails:
```
Comment (betatim) [inline FrameworkDesignConsiderations.md]: want -> what

**Comment by betatim** [comment]
Beyond the want what mixup this looks good to me. I'll merge it after that is fixed if you are happy to.

**Pull request merged**